mat64/dense.go

```diff
@@ -5,7 +5,10 @@
 package mat64
 
 import (
+	"bytes"
+
 	"github.com/gonum/blas/blas64"
+	"github.com/gonuts/binary"
```

Comment:
gonuts/binary is excessive - it provides support for a variety of types where we only need []float64. All we need to do is write an initial int64 and then the following data (in whatever format). It fosters a leave it to someoe else attitude that (below) gives us what I think is the wrong behaviour. An encode helper function would do the necessary prelude and have the correct behaviour.